base/math.jl

```diff
@@ -141,15 +144,18 @@ function evalpoly(z::Complex, p::Tuple)
         _evalpoly(z, p)
     end
 end
-evalpoly(z::Complex, p::Tuple{<:Any}) = p[1]
 
+evalpoly(z::Complex, p::Tuple{}) = zero(z)
+
+evalpoly(z::Complex, p::Tuple{<:Any}) = p[1]
 
 evalpoly(z::Complex, p::AbstractVector) = _evalpoly(z, p)
 
 function _evalpoly(z::Complex, p)
     Base.require_one_based_indexing(p)
-    length(p) == 1 && return p[1]
     N = length(p)
+    N == 0 && return zero(z)
```

Comment:
```suggestion
    N == 0 && return zero(z) * zero(eltype(p))
```
for improved type-stability? Although that will throw if `zero(eltype(p))` is undefined, e.g. if `p == Any[]`, so not sure this is better.

Reply:
Thank you for your suggestion, but I personally think this change should not be made due to the possible undefined error in `zero(eltype(p))` as you pointed out.